### PR TITLE
Various cleanups

### PR DESCRIPTION
**Pull request by dain**
## Description

Various cleanups from recent work.

## Documentation

(X) No documentation is needed.
( ) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

(X) No release notes entries required.
( ) Release notes entries required with the following suggested text:

### CODE REVIEW

File: core/trino-main/src/main/java/io/trino/connector/CatalogName.java [hunk 1]
```diff
@@ -44,6 +44,17 @@ public String getCatalogName()
         return catalogName;
     }
 
+    public CatalogName getActualName()
```
Comment (losipiuk) [inline core/trino-main/src/main/java/io/trino/connector/CatalogName.java:47]: getParentName?

File: core/trino-main/src/main/java/io/trino/connector/ConnectorManager.java [hunk 2]
```diff
@@ -616,6 +613,10 @@ public MaterializedConnector(CatalogName catalogName, Connector connector, Runna
             List<PropertyMetadata<?>> analyzeProperties = connector.getAnalyzeProperties();
             requireNonNull(analyzeProperties, format("Connector '%s' returned a null analyze properties set", catalogName));
             this.analyzeProperties = ImmutableList.copyOf(analyzeProperties);
+
+            Set<ConnectorCapabilities> capabilities = connector.getCapabilities();
+            requireNonNull(capabilities, format("Connector '%s' returned a null capabilities set", catalogName));
```
Comment (losipiuk) [inline core/trino-main/src/main/java/io/trino/connector/ConnectorManager.java:618]: nit:
```
requireNonNull(capabilities, () -> format("Connector '%s' returned a null capabilities set", catalogName));
```

File: core/trino-main/src/main/java/io/trino/metadata/CatalogMetadata.java [hunk 3]
```diff
@@ -36,110 +35,121 @@
...
+        informationSchemaTransaction.commit();
+        systemTransaction.commit();
+        catalogTransaction.commit();
+    }
+
+    public void abort()
+    {
+        safeAbort(informationSchemaTransaction);
+        safeAbort(systemTransaction);
+        safeAbort(catalogTransaction);
```
Comment (losipiuk) [inline core/trino-main/src/main/java/io/trino/metadata/CatalogMetadata.java:137]: should you throw from top level `abort` (at the end) if there was an exception in one of `safeAbort` calls?

File: core/trino-main/src/main/java/io/trino/connector/system/jdbc/ColumnJdbcTable.java [hunk 4]
```diff
@@ -174,7 +174,7 @@ public TupleDomain<ColumnHandle> applyFilter(ConnectorSession connectorSession,
             return tupleDomain;
         }
 
-        List<String> catalogs = listCatalogs(session, metadata, accessControl, catalogFilter).keySet().stream()
+        List<String> catalogs = MetadataListing.listCatalogNames(session, metadata, accessControl, catalogFilter).stream()
```
Comment (losipiuk) [inline core/trino-main/src/main/java/io/trino/connector/system/jdbc/ColumnJdbcTable.java:177]: nit: static import

File: core/trino-main/src/main/java/io/trino/connector/system/jdbc/SchemaJdbcTable.java [hunk 5]
```diff
@@ -70,7 +70,7 @@ public RecordCursor cursor(ConnectorTransactionHandle transactionHandle, Connect
         Optional<String> catalogFilter = tryGetSingleVarcharValue(constraint, 1);
 
         Builder table = InMemoryRecordSet.builder(METADATA);
-        for (String catalog : listCatalogs(session, metadata, accessControl, catalogFilter).keySet()) {
+        for (String catalog : MetadataListing.listCatalogNames(session, metadata, accessControl, catalogFilter)) {
```
Comment (losipiuk) [inline core/trino-main/src/main/java/io/trino/connector/system/jdbc/SchemaJdbcTable.java:73]: nit: static import

File: core/trino-main/src/main/java/io/trino/connector/system/jdbc/TableJdbcTable.java [hunk 6]
```diff
@@ -99,7 +99,7 @@ public RecordCursor cursor(ConnectorTransactionHandle transactionHandle, Connect
             return table.build().cursor();
         }
 
-        for (String catalog : listCatalogs(session, metadata, accessControl, catalogFilter).keySet()) {
+        for (String catalog : MetadataListing.listCatalogNames(session, metadata, accessControl, catalogFilter)) {
```
Comment (losipiuk) [inline core/trino-main/src/main/java/io/trino/connector/system/jdbc/TableJdbcTable.java:102]: nit: static import

File: core/trino-main/src/main/java/io/trino/transaction/TransactionManagerModule.java [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+{
+    @Override
+    protected void setup(Binder binder)
+    {
+        ServerConfig serverConfig = buildConfigObject(ServerConfig.class);
+        if (serverConfig.isCoordinator()) {
+            install(new InMemoryTransactionManagerModule());
+        }
+        else {
+            binder.bind(TransactionManager.class).to(NoOpTransactionManager.class).in(Scopes.SINGLETON);
```
Comment (losipiuk) [inline core/trino-main/src/main/java/io/trino/transaction/TransactionManagerModule.java:32]: You should drop this binding from `WorkerModule`

**Review by losipiuk — COMMENTED** [review]
skimmed

File: core/trino-main/src/main/java/io/trino/metadata/CatalogMetadata.java (hunk 3 above)
Comment (dain) [reply core/trino-main/src/main/java/io/trino/metadata/CatalogMetadata.java:137]: Abort is not allowed to fail, and runs in a non-connected thread to any exception thrown is just lost.

File: core/trino-main/src/main/java/io/trino/transaction/TransactionManagerModule.java (hunk 7 above)
Comment (dain) [reply core/trino-main/src/main/java/io/trino/transaction/TransactionManagerModule.java:32]: Good catch!

File: core/trino-main/src/main/java/io/trino/transaction/InMemoryTransactionManager.java [hunk 8]
```diff
@@ -443,26 +437,17 @@ private synchronized CatalogMetadata getTransactionCatalogMetadata(CatalogName c
             return catalogMetadata;
         }
 
-        public synchronized ConnectorTransactionMetadata createConnectorTransactionMetadata(CatalogName catalogName, Catalog catalog)
+        public synchronized CatalogTransaction createConnectorTransactionMetadata(CatalogName catalogName, Catalog catalog)
```
Comment (electrum) [inline core/trino-main/src/main/java/io/trino/transaction/InMemoryTransactionManager.java]: Rename this method to better match the new behavior

File: core/trino-main/src/main/java/io/trino/connector/system/MaterializedViewSystemTable.java [hunk 9]
```diff
@@ -97,7 +97,7 @@ public RecordCursor cursor(
         Optional<String> schemaFilter = tryGetSingleVarcharValue(constraint, 1);
         Optional<String> tableFilter = tryGetSingleVarcharValue(constraint, 2);
 
-        listCatalogs(session, metadata, accessControl, catalogFilter).keySet().forEach(catalogName -> {
+        MetadataListing.listCatalogNames(session, metadata, accessControl, catalogFilter).forEach(catalogName -> {
```
Comment (electrum) [inline core/trino-main/src/main/java/io/trino/connector/system/MaterializedViewSystemTable.java:100]: Static import everywhere

**Review by electrum — APPROVED** [review]

**Pull request merged**